src/App.tsx

```diff
@@ -25,33 +27,54 @@ export default class App extends React.Component<AppProps, AppState> {
       return (
           <div className="App">
               <Form>
-                  <Form.Label>Assembled Instruction</Form.Label>
+                  <Form.Label>Assembled Instruction (Hex)</Form.Label>
                   <InputGroup>
                       <InputGroup.Prepend>
                           <InputGroup.Text>0x</InputGroup.Text>
                       </InputGroup.Prepend>
-                      <FormControl onChange={(evt) => this.setState({instruction: evt.target.value})} />
+                      <FormControl onChange={(evt) => this.setState({instructionHex: evt.target.value})} />
                       <InputGroup.Append>
-                          <Button onClick={this.handleDisassemble}>Disassemble!</Button>
+                          <Button onClick={this.handleDisassembleHex}>Disassemble!</Button>
                       </InputGroup.Append>
                   </InputGroup>
-                  <Form.Text>{this.state.disassembled}</Form.Text>
+                  <Form.Text>{this.state.disassembledHex}</Form.Text>
+              </Form>
+              <Form>
+                  <Form.Label>Assembled Instruction (Binary)</Form.Label>
```

Comment:
This is probably fine for now, but something to consider might be to use a dropdown to select a base, rather than needing two different input boxes. React Bootstrap should have a Dropdown component that would work, or even just a simple form dropdown would probably work. The latter might fit in better, with this being a part of an InputGroup

Reply:
> This is probably fine for now, but something to consider might be to use a dropdown to select a base, rather than needing two different input boxes. React Bootstrap should have a Dropdown component that would work, or even just a simple form dropdown would probably work. The latter might fit in better, with this being a part of an InputGroup

I personally prefer it this way for myself but if you want to go that route I think it would be equally good. When I use tools like this online, I tend to prefer those that have several input field, so I don't have to select from a dropdown. A radio button would be a nice compromise so that you can always see the choices without having to click twice.

Reply:
I'd just like to say that Alec mentioned a dropdown box for this, without my influencing that (or, heck, even saying anything beyond "Sam made an MR that lets you use binary instead of hex")...